src/auto_renewal.cairo

```diff
@@ -116,13 +121,21 @@ mod AutoRenewal {
         naming_addr: ContractAddress,
         erc20_addr: ContractAddress,
         tax_addr: ContractAddress,
-        admin_addr: ContractAddress
+        admin_addr: ContractAddress,
+        whitelisted_renewer: ContractAddress
     ) {
         self.naming_contract.write(naming_addr);
         self.erc20_contract.write(erc20_addr);
         self.tax_contract.write(tax_addr);
         self.admin.write(admin_addr);
+        self.whitelisted_renewer.write(whitelisted_renewer);
         self.can_renew.write(true);
+        // allowing naming 2^251-1, aka infinite approval according to its implementation
+        // when moving funds, the storage variable won't be updated, saving gas
+        IERC20CamelDispatcher { contract_address: erc20_addr }
+            .approve(
+                naming_addr, 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
```

Comment:
I think you can use `integer::BoundedInt::max()` here

Reply:
great tip